File: php/tests/NoAuthTest.php
<?php

namespace Tests\Auth;

use Auth\NoAuth;

class NoAuthTest extends TestUtils
{
    private $auth;

    protected function setUp(): void
    {
        parent::setUp();
        $this->auth = new NoAuth();
    }

    public function testNoAuthenticationRequired(): void
    {
        $request = $this->createTestRequest('GET');
        $this->runAuthTest($this->auth, $request, 200, false);
    }

    public function testRequestWithHeaders(): void
    {
        $request = $this->createTestRequest('GET', [
            'Authorization' => 'Bearer test-token'
        ]);
        $this->runAuthTest($this->auth, $request, 200, false);
    }

    public function testRequestWithQueryParameters(): void
    {
        $request = $this->createTestRequest('GET', [
            'X-API-Key' => 'test-api-key'
        ]);
        $this->runAuthTest($this->auth, $request, 200, false);
    }

    public function testGetType(): void
    {
        $this->assertAuthType($this->auth, 'none');
    }
} 

File: java/src/test/java/com/auth/NoAuthTest.java
package com.auth;

import com.auth.base.Authentication;
import com.auth.none.NoAuth;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.http.HttpMethod;
import org.springframework.http.HttpStatus;
import org.springframework.http.RequestEntity;

import static org.junit.jupiter.api.Assertions.*;

class NoAuthTest extends TestUtils {
    private Authentication auth;

    @BeforeEach
    void setUp() {
        super.setUp();
        auth = new NoAuth();
    }

    @Test
    void testNoAuthenticationRequired() {
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, null);
        runAuthTest(auth, request, HttpStatus.OK, false);
    }

    @Test
    void testGetType() {
        assertAuthType(auth, "none");
    }
} 